In [ ]:
from sklearn.datasets import load_diabetes
from sklearn.linear_model import LinearRegression
from sklearn import metrics
from sklearn.model_selection import train_test_split
import matplotlib.pyplot as plt

In [ ]:
X, y = load_diabetes(return_X_y=True)

# Regressor with default test_size(0,25)

In [ ]:
X_train, X_test, y_train, y_test = train_test_split(X, y, shuffle=True)

In [ ]:
regressor = LinearRegression().fit(X_train, y_train)

In [ ]:
regressor.score(X_test, y_test)

0.47533210948729365

In [ ]:
y_pred = regressor.predict(X_test)

In [ ]:
y_test

array([ 85., 182.,  96.,  98., 243., 178.,  94., 108., 104., 156., 229.,
       146., 222., 109., 242., 150., 214.,  40.,  66.,  84., 242.,  81.,
       129., 177.,  71., 167., 268.,  72., 310., 148., 248.,  97., 196.,
        96., 191.,  63.,  42.,  63., 131., 107., 262., 170., 181., 244.,
       200.,  69., 257., 268., 259., 160., 210., 114., 163.,  70., 109.,
       302., 196.,  71.,  90., 129.,  68.,  77., 252., 128., 101.,  53.,
       144., 150., 109., 128., 230., 265., 122., 180.,  73., 237., 142.,
       147., 270., 200., 232., 170., 190.,  94., 209.,  51.,  92., 151.,
        71., 111., 288.,  72., 143., 179.,  57., 195.,  61., 125., 198.,
       180., 185.,  65., 127., 236.,  93.,  48., 215., 121., 275., 245.,
        55.])

In [ ]:
y_pred

array([ 53.10052886, 134.57149584, 108.18009441,  79.4572534 ,
       255.58118434, 129.60428151, 162.83214095, 107.3988554 ,
       101.93036148, 163.48475031, 185.56519867, 147.92629076,
       198.97442611, 166.19346225, 166.27433726, 116.84828222,
       117.08296249, 140.80667523, 180.09831616,  89.27283029,
       285.84467953,  96.83237074, 197.61696088, 123.35419662,
        79.13577062, 185.8598474 , 218.082152  ,  39.90988933,
       260.23005531, 130.27417586, 223.44945781, 146.60387662,
       162.90801921,  52.15636952, 118.93823451, 120.90293423,
       129.27470272,  99.03390942, 166.34553304, 114.41828974,
       177.77224508, 196.26063058,  83.25297271, 181.93422452,
       118.09349522,  96.21147984, 194.80365201, 229.84671963,
       233.98366077, 110.80613553, 148.42529515,  84.38271315,
       214.94353819,  53.85072529, 152.39917748, 141.68011682,
       159.22190835, 127.0579976 , 178.61821898, 153.42052971,
       120.16780133,  55.3020852 , 156.63179518, 225.91

In [ ]:
metrics.mean_absolute_error(y_test, y_pred)

41.902410572830746

Regressor mean absolute error = 41.9

# Regressor with test_size = 0,33

In [ ]:
X_train_2, X_test_2, y_train_2, y_test_2 = train_test_split(X, y, test_size=0.33, shuffle=True)

In [ ]:
regressor_2 = LinearRegression().fit(X_train_2, y_train_2)

In [ ]:
regressor_2.score(X_test_2, y_test_2)

0.4666780408550254

Regressor score went down with test_size=0.33

In [ ]:
y_pred_2 = regressor_2.predict(X_test_2)

In [ ]:
y_test_2

array([220., 288.,  55., 185., 233., 144., 208., 265., 190.,  65., 262.,
       161., 132., 116., 259., 140., 268., 118.,  68., 212., 142., 233.,
       280.,  68., 141., 232., 143., 259., 252.,  70.,  85., 261.,  60.,
       274.,  78., 152., 127.,  93., 153., 109.,  96., 308., 200., 242.,
       199.,  75.,  54., 150., 131., 192.,  64.,  91., 241., 109., 129.,
       104.,  49., 135., 102.,  84., 175., 120., 160., 178., 158., 202.,
       152., 141.,  65., 236.,  83., 162.,  50.,  83., 150.,  95., 118.,
       200., 200.,  68., 167.,  90.,  63., 163.,  53., 310., 259., 346.,
       263., 237.,  89., 136., 206., 156.,  40., 216.,  78., 128., 202.,
       220., 235., 198., 128.,  91., 166., 124., 310.,  69., 180., 104.,
       195.,  67., 174.,  86.,  96.,  93., 264., 143., 132.,  44., 158.,
        59., 179., 225., 126.,  58., 265., 230.,  49., 243.,  88., 172.,
       219., 252., 276., 185., 268., 113.,  69., 168.,  78., 220., 237.,
       181., 178.,  63.])

In [ ]:
y_pred_2

array([201.61251977, 204.76628445,  70.71360625, 140.86396931,
       201.83345514, 179.70905546, 229.95072683, 196.50384726,
       175.34015195, 128.62548711, 159.7236546 , 190.64712143,
       119.51331238,  48.88207691, 160.46783282, 171.71532617,
       234.21246636, 154.5620123 , 118.21541236, 194.35599567,
       103.59087559, 263.99811919, 232.85313577, 200.93495228,
       168.63558061, 227.29682593, 171.96488338, 218.658613  ,
       253.11174729,  59.86572074,  57.52373294, 228.56015889,
        70.83247748, 251.46220242,  55.39879603, 244.75548353,
       111.01831733,  79.6146235 , 112.91936086, 182.36678223,
       120.50241065, 268.52522146, 156.1858595 , 303.81218054,
       103.65860632,  71.72209268,  93.52983167, 207.73185015,
       153.34714336, 224.26416605,  85.55576073,  89.06918322,
       205.22077705, 105.18515851, 188.97053763, 158.98168315,
       130.0480245 , 126.47123065,  82.15662595, 186.46118822,
       185.5957125 , 166.57478232, 116.82670051, 193.66

In [ ]:
metrics.mean_absolute_error(y_test_2, y_pred_2)

44.37702238420008

Regressor mean absolute error is greater with test_size=0.33